motioneye/mjpgclient.py

```diff
@@ -157,6 +157,11 @@ def _error(self, error) -> None:
 
     def _on_connect(self, future: Future) -> None:
         try:
+            if self.closed():
+                future.cancel()
+                logging.debug("_on_connect: Stream is closed. Future cancelled.")
+                return
```

Comment:
Do we want to move this into an own internal function? Although it wouldn't reduce much code, since `return` and passing `future` and the function name needs to be done here, but it would allow to change the remaining log text in one place... not so much benefit (yet), on a second though.

Reply:
Yeah thought it too and didn't see any benefits(yet) :)

Reply:
Ok, added _get_future_result method and maybe that reduced code a bit.